webapp-django/crashstats/crashstats/static/crashstats/js/socorro/oauth2.js

```diff
@@ -0,0 +1,134 @@
+var OAuth2 = (function() {
+    /* Generally the best documentation for working with GoogleAuth
+    is here: https://developers.google.com/identity/sign-in/web/reference
+    The introduction is here:
+    https://developers.google.com/identity/sign-in/web/sign-in
+    */
+
+    var client_id = document.head.querySelector(
+        'meta[name="google-signin-client_id"]'
+    ).content;
+
+    var GoogleAuth = null;
+
+    var loadGoogleAuth = function() {
+        return new Promise(function(resolve, reject) {
+            if (!GoogleAuth) {
+                gapi.auth2.init();
+                GoogleAuth = gapi.auth2;
+            }
+            // If you instead try to do:
+            // resolve(GoogleAuth.getAuthInstance()) here inside
+            // this promise body your browser will crash. Both
+            // Firefox and Chrome.
+            resolve(GoogleAuth);
+        });
+    };
+
+    var serverSignout = function() {
+        var signoutURL = $('a.google-signout').attr('href');
+        var csrfmiddlewaretoken = $(
+            'input[name="csrfmiddlewaretoken"]'
+        ).val();
+        var data = {
+            csrfmiddlewaretoken: csrfmiddlewaretoken,
+        };
+        $.post(signoutURL, data)
+        .done(function() {
+            // XXX could this be smarter
+            // https://bugzilla.mozilla.org/show_bug.cgi?id=1283296
+            // For example, you might be on a sensitive URL
+            // so redirect will just become a redirect fest.
+            document.location = '/';
```

Comment:
I believe this is what we do at the moment with persona sign out as well, so this is fine by me. It's a lot of trouble to redirect somewhere different than home when signing out, for very low value.